setup.py

```diff
@@ -139,6 +140,24 @@
     "soundfile",
     "transformers",
     "zstandard",
+]
+
+TEST_PY37_ONLY = [
+    "apache-beam",  # constrains the dill version, which constrains the multiprocess version too much (incompatible with py3.10)
+    "aiobotocore",  # TODO: support the S3 tests on py3.10
+    "boto3",
+    "botocore",
+    "moto",
+    "s3fs"
+]
+
+
+for i, dependency in enumerate(TESTS_REQUIRE):
+    if re.match(r"[\w-]+", dependency).group() in TEST_PY37_ONLY:
+        TESTS_REQUIRE[i] = dependency + ";python_version<'3.8'"
```

Comment:
Why don't adding ";python_version<'3.8'" directly to these elements in `TESTS_REQUIRE` and avoiding a regex here?

Reply:
just removed it :)